tests/test_updater_with_simulator.py

```diff
@@ -152,6 +152,33 @@ def test_keys_and_signatures(self):
 
         self._run_refresh()
 
+    def test_snapshot_rollback_check_with_missing_local_snapshot(self):
+        # Test triggering snapshot rollback check on a newly downloaded snapshot
+        # when the local snapshot is not loaded due to hash mismatch.
+        # Initialize all metadata and assign targets version higher than 1.
+        self.sim.targets.version = 2
+        self.sim.update_snapshot()
+        self._run_refresh()
+
+        # The new targets should have a lower version than the local trusted one.
+        self.sim.targets.version = 1
+        self.sim.update_snapshot()
+
+        # Calculate the hash of the new modified version of snapshot.
+        digest_object = sslib_hash.digest("sha256")
+        digest_object.update(self.sim._fetch_metadata("snapshot"))
+        new_snapshot_hash = digest_object.hexdigest()
+        # Assign the new hash to meta in timestamp, so that loading the local
+        # snapshot will fail, but loading the next snapshot version will succeed.
+        self.sim.timestamp.snapshot_meta.hashes = {"sha256": new_snapshot_hash}
```

Comment:
looking at this with fresh eyes, this seems out of place here: our tests should not be this complex. I don't know where the meta file hashing itself should eventually happen but the test should not have to do any of this... The repo (in this case RepositorySimulator) should instead support `meta` that contains length and hashes.

I think the test should just have something like  `self.sim.metafile_hashes = True` in the beginning of the test and then `self.sim.update_snapshot()` should just work without any manual modification. 

This means `RepositorySimulator.update_snapshot()` (and `update_timestamp()`) would include hashes/length if the `metafile_hashes` attribute is set.

Can you try if this approach is sensible?